jolt-core/src/lasso/memory_checking.rs

```diff
@@ -77,10 +77,8 @@ where
     Polynomials: StructuredCommitment<C>,
     Self: std::marker::Sync,
 {
-    type ReadWriteGrandProduct: BatchedGrandProduct<F> + Send + 'static =
-        BatchedDenseGrandProduct<F>;
-    type InitFinalGrandProduct: BatchedGrandProduct<F> + Send + 'static =
-        BatchedDenseGrandProduct<F>;
+    type ReadWriteGrandProduct: BatchedGrandProduct<F, C> + Send + 'static = QuarkGrandProduct<F>;
+    type InitFinalGrandProduct: BatchedGrandProduct<F, C> + Send + 'static = QuarkGrandProduct<F>;
```

Comment:
On second thought, let's keep using `BatchedDenseGrandProduct` as the default –– I don't want someone benchmarking Jolt to think we've had a performance regression based on running the Quark grand product with Hyrax